'Go' build dependencies error while installing m1-terraform-provider-helper install hashicorp/vault -v v2.2.0

<!-- Please replace Failing to Install m1-terraform-provider-helper install hashicorp/vault -v v2.2.0 due to go dependencies with your description -->


## Expected Behavior

Be able to initialize my  terraform project

## Actual Behavior

Can't initialize my terraform project
![image](https://user-images.githubusercontent.com/104012132/175093798-c1ccab61-b635-466c-a143-1d72963eef27.png)


## Steps to Reproduce (including precondition)

1. m1-terraform-provider-helper activate
2. m1-terraform-provider-helper install hashicorp/vault -v v2.2.0


## Screenshot on This Problem (if possible)

```
m1-terraform-provider-helper install hashicorp/vault -v v2.2.0
Repo: https://github.com/hashicorp/terraform-provider-vault
GitRepo: https://github.com/hashicorp/terraform-provider-vault
HEAD is now at 3e8371ce Wait for Vault to garbage collect deleted namespaces (#1506)
2022/06/22 13:37:11 Pulling newest changes from https://github.com/hashicorp/terraform-provider-vault
fatal: ref refs/remotes/origin/HEAD is not a symbolic ref
Already up to date.
2022/06/22 13:37:11 version: v2.2.0
==> Checking that code complies with gofmt requirements...
go install
../../go/pkg/mod/cloud.google.com/go@v0.39.0/iam/iam.go:29:2: missing go.sum entry needed to verify package github.com/googleapis/gax-go/v2 (imported by cloud.google.com/go/storage) is provided by exactly one module; to add:
	go get cloud.google.com/go/storage@v0.39.0
make: *** [build] Error 1
2022/06/22 13:37:18 Bash code did not run successfully: exit status 2
```
![image](https://user-images.githubusercontent.com/104012132/175094433-4990eba1-270f-405d-a62e-4c45f94452c0.png)


## Your Environment

- OS: MacOS Monterrey 12.4
- m1-terraform-provider-helper version: v0.6.1
- go version go1.18.3 darwin/arm64

--- a/internal/app/install.go
+++ b/internal/app/install.go
@@ -50,7 +50,11 @@ func executeBashCommand(command string, baseDir string) {
 	}
 
 	if err := cmd.Run(); err != nil {
-		log.Fatalf("Bash code did not run successfully: %s", err)
+		if exitError, ok := err.(*exec.ExitError); ok {
+			if exitError.ExitCode() != 0 {
+				log.Fatalf("Bash execution did not run successfully: %s", err)
+			}
+		}
 	}
 }
 
@@ -125,7 +129,7 @@ func checkoutSourceCode(baseDir string, gitURL string, version string) string {
 	// Clean the repository
 	executeBashCommand("git reset --hard && git clean -d -f -q", fullPath)
 	log.Println("Pulling newest changes from " + gitURL)
-	executeBashCommand("git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@' | xargs git checkout && git pull", fullPath)
+	executeBashCommand("git remote show origin | sed -n '/HEAD branch/s/.*: //p'| xargs git checkout && git pull", fullPath)
 
 	if len(version) > 0 {
 		log.Println("version: " + version)

--- a/internal/app/install.go
+++ b/internal/app/install.go
@@ -3,6 +3,7 @@ package app
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io/ioutil"
 	"log"
@@ -50,8 +51,9 @@ func executeBashCommand(command string, baseDir string) {
 	}
 
 	if err := cmd.Run(); err != nil {
-		if exitError, ok := err.(*exec.ExitError); ok {
-			if exitError.ExitCode() != 0 {
+		var e *exec.ExitError
+		if errors.As(err, &e) {
+			if e.ExitCode() != 0 {
 				log.Fatalf("Bash execution did not run successfully: %s", err)
 			}
 		}

--- a/internal/app/install_test.go
+++ b/internal/app/install_test.go
@@ -109,3 +109,15 @@ func TestGetProviderData(t *testing.T) {
 		}
 	})
 }
+
+func TestCheckoutSourceCode(t *testing.T) {
+	t.Run("Should checkout source code once", func(t *testing.T) {
+		tmpDir := t.TempDir()
+		checkoutSourceCode(tmpDir+"/cliCache", "https://github.com/hashicorp/terraform-provider-random", "v2.2.0")
+	})
+	t.Run("Should checkout two versions of same source code once", func(t *testing.T) {
+		tmpDir := t.TempDir()
+		checkoutSourceCode(tmpDir+"/cliCache", "https://github.com/hashicorp/terraform-provider-random", "v2.3.1")
+		checkoutSourceCode(tmpDir+"/cliCache", "https://github.com/hashicorp/terraform-provider-random", "v2.2.0")
+	})
+}

